java-squid/src/main/java/org/sonar/java/se/checks/UnclosedResourcesCheck.java

```diff
@@ -61,20 +64,24 @@
 @SqaleConstantRemediation("5min")
 public class UnclosedResourcesCheck extends SECheck implements JavaFileScanner {
 
-  private enum States {
+  private enum Status {
     OPENED, CLOSED;
   }
 
-  private static final String JAVA_IO_CLOSEABLE = "java.io.Closeable";
+  private static final String JAVA_IO_CLOSEABLE = "java.lang.AutoCloseable";
```

Comment:
There is an inconsistency between name and value of the constant. 
Moreover looking only for subtype of AutoCloseable is not correct as this interface was introduced in java 1.7 https://docs.oracle.com/javase/7/docs/api/java/lang/AutoCloseable.html and Closeable is available since 1.5 https://docs.oracle.com/javase/7/docs/api/java/io/Closeable.html 

This would work as those are part of JDK and we require java 7 for analysis and thus any java.io.Closeable will be indirectly inherited from AutoCloseable but this should not be relied upon as we want to move away from that dependency on the JDK doing the analysis in semantic. 

So we should detect AutoCloseable and if not match -> Closeable.